cmd/signature-v4.go

```diff
@@ -146,6 +147,24 @@ func doesPolicySignatureMatch(formValues http.Header) APIErrorCode {
 	return doesPolicySignatureV4Match(formValues)
 }
 
+// compareSignatureV4 returns true if and only if both signatures
+// are equal. The signatures are excepted to be HEX encoded strings
+// according to the AWS S3 signature V4 spec.
+func compareSignatureV4(sig1, sig2 string) bool {
+	// convert signature to binary representation
+	// subtle.ConstantTimeCompare([]byte(sig1), []byte(sig2)) is not
+	// recommended even though it would work for HEX values.
```

Comment:
I suggest scrapping this comment, and the unnecessary decoding of hexadecimal to binary. Just add a comment that since hexadecimal encodings are unique for a sequence of bytes, decoding is not required.